### PR TITLE
added Regular Component - Method Component section

### PR DESCRIPTION
**Pull request by khudiakov**

### CODE REVIEW

File: src/pages/ClassComponent.mdx [hunk 1]
```diff
@@ -176,3 +176,92 @@ export class ChildClassComponentWithObjectPropsMemoized extends Component<
 but when parent component changes `obj.str` re-rendering happens, as expected:
 
 <ObjectPropsClassComponent isMemoized={true} changeProps={true} />
+
+---
+
+### Regular Component - Method Component
```
Comment (mifrdlik) [inline src/pages/ClassComponent.mdx]: Do people really call it method components?

Comment (smykhailov) [reply src/pages/ClassComponent.mdx]: I'm not sure if it has any official name. If yes, we can use it, if not, this term explains what it is about

File: src/pages/ClassComponent.mdx [hunk 2]
```diff
@@ -176,3 +176,92 @@ export class ChildClassComponentWithObjectPropsMemoized extends Component<
...
+
+Method Component is a component which was implemented as a **method of Class Component**.
```
Comment (smykhailov) [inline src/pages/ClassComponent.mdx]: Lets extract this to the separate page - Method Component
and add interactive example if it illustrates re-rendering

File: src/pages/ClassComponent.mdx (hunk 1 above)
Comment (khudiakov) [reply src/pages/ClassComponent.mdx]: Yeah, I just invented it 😄 Will create an article in Wiki

**Pull request merged**